code/game/objects/items/food/pastries.dm

```diff
@@ -357,7 +357,26 @@
 	tastes = list("muffin" = 3, "spookiness" = 1)
 	foodtypes = GRAIN | FRUIT | SUGAR | BREAKFAST
 
+/obj/item/food/muffin/moffin
+	name = "moffin"
+	icon_state = "moffin"
+	desc = "A delicious and spongy little cake."
+	tastes = list("muffin" = 3, "dust" = 1, "lint" = 1)
+	foodtypes = CLOTH | GRAIN | SUGAR | BREAKFAST
+
+/obj/item/food/muffin/moffin/Initialize(mapload)
+	. = ..()
+	icon_state = "[icon_state]_[rand(1,3)]"
 
+/obj/item/food/muffin/moffin/examine(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/moffin_observer = user
+	if(foodtypes & moffin_observer.dna.species.liked_food)
```

Comment:
I don't think this is correct--if someone who likes grain, sugar, or breakfast looks at it they'll still get this text. I think you just want to check `liked_food & CLOTH`.